meta-client/src/main/java/org/apache/helix/metaclient/factories/MetaClientFactory.java

```diff
@@ -32,6 +35,12 @@ public class MetaClientFactory {
   private static final Logger LOG = LoggerFactory.getLogger(MetaClientFactory.class);
 
   public MetaClientInterface getMetaClient(MetaClientConfig config) {
+    if (config.getStoreType() == MetaClientConfig.StoreType.ZOOKEEPER) {
+      ZkMetaClientConfig zkMetaClientConfig = new ZkMetaClientConfig.ZkMetaClientConfigBuilder().
```

Comment:
If we create a new zkMetaClientConfig and only set the `getConnectionAddress`, will other settings lost?

Reply:
Yes good catch! I'll set all configurable fields from metaclientconfig when creating a zkmetaclientconfig